docs/javascript/v2/guides/javascript-quickstart.mdx

```diff
@@ -310,85 +244,42 @@ muteVid.addEventListener("click", () => {
 
-If you try it out now, you may notice that turning off the video makes the tile go blank but turning it on again doesn't bring it back.
-This is due to a change in video track requiring a call to attach again. To fix it let's
-call our `renderPeers` function whenever we change the state of the local video.
-
-```jsx{8}:script.js
-muteVid.addEventListener("click", () => {
-  const videoEnabled = hmsStore.getState(selectIsLocalVideoEnabled);
-
-  hmsActions.setLocalVideoEnabled(!videoEnabled);
-
-  muteVid.textContent = !videoEnabled ? "Hide" : "Unhide";
-
-  renderPeers(hmsStore.getState(selectPeers));
-});
-```
-
 ![Mute/Unmute](/guides/mute-unmute.gif)
 
-## Mirroring local video
-
-Each `peer` object has an `isLocal` property that indicates whether the peer is local or remote.
-When looping over the peer list we can use this property to decide whether to mirror the video or not.
+### Changing UI based on connection state
 
-<Tabs id="mirror" items={['JS', 'CSS']} />
-
-<Tab id="mirror-0" >
+Right now, our join form shows even after we have joined the room. We need a way to know the connection state of the room and hide the form after we've connected.
 
-```jsx{10,28}:script.js
-function renderPeers(peers) {
-  peersContainer.innerHTML = "";
+We can do this by subscribing to the store with the `selectIsConnectedToRoom` selector.
 
-  if (!peers) {
-    peers = hmsStore.getState(selectPeers);
+```jsx:script.js
+//Showing the required elements on connection/disconnection
+function onConnection(isConnected) {
+  if (isConnected) {
+    form.classList.add("hide");
+    conference.classList.remove("hide");
+    leaveBtn.classList.remove("hide");
+    controls.classList.remove("hide");
+  } else {
+    form.classList.remove("hide");
+    conference.classList.add("hide");
+    leaveBtn.classList.add("hide");
+    controls.classList.add("hide");
   }
-
-  peers.forEach((peer) => {
-    const video = h("video", {
-      class: "peer-video" + (peer.isLocal ? " local" : ""),
-      autoPlay: true,
-      muted: true
-    });
-
-    hmsActions.attachVideo(peer.videoTrack, video);
-
-    const peerContainer = h(
-      "div",
-      {
-        class: "peer-container"
-      },
-      video,
-      h(
-        "div",
-        {
-          class: "peer-name"
-        },
-        peer.name + (peer.isLocal ? " (You)" : "")
-      )
-    );
-
-    peersContainer.append(peerContainer);
-  });
 }
-```
-
-</Tab>
 
-<Tab id="mirror-1">
-
-```css:styles.css
-.local.peer-video {
-    transform: scaleX(-1);
-}
+// listen to the connection state
+hmsStore.subscribe(onConnection, selectIsConnectedToRoom);
 ```
 
-</Tab>
+## Test the app
 
-That wraps it up for this guide. We hope you had fun.
-There are plenty of cool stuff which can be done with our SDK, be sure to check the [features section](/javascript/v2/features/integration) for more information.
+
+## Next steps
 
 Here's the complete example.
 
 <Codesandbox id="5rmes" />
+
```

Comment:
seeing a broken image on the top left of the sandbox 

<img width="327" alt="Screenshot 2022-07-02 at 4 17 59 PM" src="https://user-images.githubusercontent.com/284612/177018733-2bb5ffee-f5ad-44ea-9d42-cddd40579955.png">